tests/test_pygtracker.py

```diff
@@ -22,6 +33,34 @@ def test_version():
 
 # Start tests for rank_students
 
+# tests that the user inputs the right number of students
+def test_number_of_students():
+    assert (
+        gradetracker.rank_students(n=4).shape[0]
+    ) == 4, "Number of students don't match"
+
+
+# raises an error when a course id is not a part of the dataset
+def test_course_input():
+    with raises(NameError):
+        gradetracker.rank_students(course_id="500", n=4)
+
+
+# dataframe is equal to an example dataframe
+def test_df_equal():
+    example_df = pd.DataFrame(
+        {"student_id": "nameofstudent", "rank": 1, "grade": "addgradehere"}, index=[1]
+    )
+    assert example_df.equals(gradetracker.rank_students(course_id="511", n=1))
+
+
+# check the columns of the dataframe that come of the function
+def test_df_columns():
```

Comment:
It's better to run the rank_students() once, assign to a variable, and then call that variable 3 times

student_ranks = gradetracker.rank_students(course_id="511", n=1)
assert rank.columns[0] == ....